src/Credentials/CredentialProvider.php

```diff
@@ -705,6 +705,12 @@ private static function loadRoleProfile(
         $sourceProfileName = "";
         if (!empty($roleProfile['source_profile'])) {
             $sourceProfileName = $roleProfile['source_profile'];
+            # in ~/.aws/config all the named profile (except 'default') are
+            # prefix with 'profile '
+            if ($filename == (self::getHomeDir() . '/.aws/config') &&
+                $sourceProfileName != 'default') {
+                $sourceProfileName = 'profile ' . $sourceProfileName;
```

Comment:
Would you please be able to add a check here for whether `profile ` has been already  appended to the profile. By doing this we keep backward compatibility for customers that are already adding the `profile ` to profile names.